### PR TITLE
Fix bug in kube-proxy of not updating iptables rules if a service's public IPs change

### PR DESCRIPTION
**Pull request by a-robinson**
And add tests to catch the bug. Ran into this while testing #5180.

### CODE REVIEW

**Comment by a-robinson** [comment]
Updated based on suggestion in #6061.

File: pkg/proxy/proxier_test.go [hunk 1]
```diff
@@ -551,6 +551,77 @@ func TestUDPProxyUpdatePort(t *testing.T) {
...
+	if !exists {
+		t.Fatalf("can't find serviceInfo")
+	}
+	testEchoTCP(t, "127.0.0.1", svcInfo.proxyPort)
+	// This is a bit async, but this should be sufficient.
+	time.Sleep(500 * time.Millisecond)
+	waitForNumProxyLoops(t, p, 1)
+}
+
+func TestUDPProxyUpdatePublicIPs(t *testing.T) {
```
Comment (thockin) [inline pkg/proxy/proxier_test.go]: Probably don't need TCP and UDP - this has nothing to do with ports, rigth?

Comment (a-robinson) [inline pkg/proxy/proxier_test.go]: Right, good point. Removed.

**Comment by quinton-hoole** [comment]
Thanks Tim.  Sorry - I just noticed this now :-(

**Pull request merged**